moped-editor/src/views/projects/projectView/ProjectComponents/ProjectSourcesAndLayers.js

```diff
@@ -0,0 +1,127 @@
+import { Source, Layer } from "react-map-gl";
+import { MAP_STYLES } from "./mapStyleSettings";
+import { useFeatureTypes } from "./utils/map";
+
+/**
+ * Component that renders feature collection of existing component features in a project
+ * All layers are set to show below basemap street labels using beforeId = "street-labels"
+ * @param {Boolean} isCreatingComponent - are we creating a component?
+ * @param {Boolean} isEditingComponent - are we editing a component?
+ * @param {String} linkMode - Tracks if we are editing "lines" or "points"
+ * @param {Object} clickedComponent - Details of the component that was clicked
+ * @param {Boolean} isDrawing - is the user currently drawing?
+ * @param {Object} projectComponentsFeatureCollection - GeoJSON feature collection with all project features
+ * @returns JSX.Element
+ */
+const ProjectSourcesAndLayers = ({
+  isCreatingComponent,
+  isEditingComponent,
+  linkMode,
+  clickedComponent,
+  isDrawing,
+  projectComponentsFeatureCollection,
+  draftEditComponent,
+}) => {
+  // If we are editing, we don't show the feature being edited
+  const draftEditComponentId = draftEditComponent?.project_component_id;
+
+  const projectFeaturesWithoutComponentBeingEdited = draftEditComponent
+    ? {
+        ...projectComponentsFeatureCollection,
+        features: projectComponentsFeatureCollection.features.filter(
+          (feature) => feature.component_id !== draftEditComponentId
+        ),
+      }
+    : projectComponentsFeatureCollection;
```

Comment:
This is the important part - we filter out showing the greyed out features of the component currently being edited. Otherwise, when we click on an orange feature during editing, it would show the greyed out and stale feature below .